src/tcache.c

```diff
@@ -756,21 +868,29 @@ tcache_create_explicit(tsd_t *tsd) {
 	 * the beginning of the whole allocation (for freeing).  The makes sure
 	 * the cache bins have the requested alignment.
 	 */
-	size_t size = tcache_bin_alloc_size + sizeof(tcache_t)
+	thread_tcache_max_and_nhbins_init(tsd);
```

Comment:
This shouldn't be necessary. When we reach here, we have a valid state `tsd` which means this was done already.

However a question: do we want the manually created tcache to use the tcache max settings of the thread created it? Or they should always start with the process default, since they may be passed around among threads?

Reply:
Yup, will remove that. 

Good question. Since we are now storing the tcache_max and nhbins in a per-thread pattern, a tcache actually does not store its own tcache_max and nhbins. Whether the given tcache is used properly depends on [`tcache_small_bin_disabled`](https://github.com/jemalloc/jemalloc/pull/2493/files#diff-4fb270192eb9659e203ae9cd89e3020592559686d14774e11eaf645b93e20738R89), which relies on the `bin_info` of the bin and only asserts on `thread_nhbins`. So I think this is good. 

Yet this reminds me that we should probably remove the [assertion](https://github.com/jemalloc/jemalloc/pull/2493/files#diff-4fb270192eb9659e203ae9cd89e3020592559686d14774e11eaf645b93e20738R96) considering this case from manually created tcache?

Reply:
The assertion link isn't working (pointing to a blank line).